addons/account/models/account_move.py

```diff
@@ -2926,7 +2926,7 @@ class AccountMoveLine(models.Model):
         help="The move of this entry line.")
     move_name = fields.Char(string='Number', related='move_id.name', store=True, index=True)
     date = fields.Date(related='move_id.date', store=True, readonly=True, index=True, copy=False, group_operator='min')
-    ref = fields.Char(related='move_id.ref', store=True, copy=False, index=True, readonly=False)
+    ref = fields.Char(related='move_id.ref', store=True, copy=False, index=False, readonly=False)
```

Comment:
Are you sure about this one?
> A multicolumn B-tree index can be used with query conditions that involve any subset of the index's columns, **but the index is most efficient when there are constraints on the leading (leftmost) columns.**

https://www.postgresql.org/docs/10/indexes-multicolumn.html

We should rename this field, this should help seeing where it is used.

Reply:
@nseinlet meant several different things, I think:

1) There *is* a multicolumn index on `(partner_id, ref)`, and this is not the reason why he thinks the btree index on `ref` should be dropped. Indeed it could still be useful to have both if we search on `ref` alone.

And yes, technically a multi-column index *could* be used even without criterions on the leading columns, but in practice that does not work because it ranks too low in the possible plans available [for the planner](https://www.postgresql.org/docs/12/planner-optimizer.html). So we can never rely on that.
Any other relevant index would definitely be used first, and possibly even the default full table scan. You can search google "why postgres won't use my index" for examples, it's a common question. The cheapest plan is not always the one you think ;-)

2) The simple btree index on `ref` seems to be entirely unused, from what we see in live databases (index scans statistics: 0). So either the business code never needs it, or it's always used with `ilike '%s<pattern>%s'` criterions (the default way the 'ilike' operator works), which won't use normal btree indexes, because there is no known prefix to scan the index tree. Either way, it takes space (and time) for nothing.

As a result, he proposes to drop it.